### PR TITLE
scripts/check_style.sh: add exclude list support.

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

File: scripts/check_style.sh [hunk 1]
```diff
@@ -2,13 +2,20 @@
...
+EXCLUDE_FILES_JOINED=$(printf "%s\|" "${EXCLUDE_FILES[@]}")
+EXCLUDE_FILES_JOINED=${EXCLUDE_FILES_JOINED%??}
+
 (
 REPO_ROOT="$(dirname "$0")"/..
 cd $REPO_ROOT
 
 WHITESPACE=$(git grep -n -I -E "^.*[[:space:]]+$" |
   grep -v "test/libsolidity/ASTJSON\|test/libsolidity/ASTRecoveryTests\|test/compilationTests/zeppelin/LICENSE" |
-  grep -v -E "test/libsolidity/syntaxTests/comments/unicode_direction_override_1.sol"
```
Comment (aarlt) [inline scripts/check_style.sh:11]: What was the reason for this?

File: scripts/check_style.sh [hunk 2]
```diff
@@ -2,13 +2,20 @@
...
+  grep -v "${EXCLUDE_FILES_JOINED}"
```
Comment (aarlt) [inline scripts/check_style.sh]: This could be combined with the line above (line 17).

File: scripts/check_style.sh (hunk 1 above)
Comment (chriseth) [reply scripts/check_style.sh:11]: I think this was an attempt at fixing something that in the end was fixed differently.

**Review by chriseth — COMMENTED** [review]
Did you check that the mechanism is still working?

**Comment by aarlt** [comment]
> Did you check that the mechanism is still working?

@chriseth yes I checked the mechanism. It seem to work.

**Review by leonardoalt — APPROVED** [review]

**Review by cameel — APPROVED** [review]
This looks fine. Test failures were completely unrelated (problems with Truffle in OpenZeppelin external tests, #10672). I retriggered them and they should be passing now.

**Pull request merged**